addons/web_editor/static/tests/banner_tests.js

```diff
@@ -0,0 +1,90 @@
+/** @odoo-module **/
+import { setSelection } from "@web_editor/js/editor/odoo-editor/src/utils/utils";
+import { makeView, setupViewRegistries } from "@web/../tests/views/helpers";
+import { patchWithCleanup, nextTick } from "@web/../tests/helpers/utils";
+import { Wysiwyg } from "@web_editor/js/wysiwyg/wysiwyg";
+import {
+    triggerEvent,
+    insertText,
+} from "@web_editor/js/editor/odoo-editor/test/utils";
+
+function onMount() {;
+    const editor = wysiwyg.odooEditor;
+    const editable = editor.editable;
+    editor.testMode = true;
+    return { editor, editable };
+}
+
+let serverData;
+let wysiwyg;
+
+QUnit.module(
+    "web_editor",
+    {
+        before: function () {
+            serverData = {
+                models: {
+                    note: {
+                        fields: {
+                            body: {
+                                string: "Editor",
+                                type: "html",
+                            },
+                        },
+                        records: [
+                            {
+                                id: 1,
+                                display_name: "first record",
+                                body: "<p><br></p>",
+                            },
+                        ],
+                    },
+                },
+            };
+        },
+        beforeEach: async function () {
+            setupViewRegistries();
+            patchWithCleanup(Wysiwyg.prototype, {
+                async init() {
+                    await this._super(...arguments);
+                    wysiwyg = this;
+                },
+            });
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "note",
+                arch:
+                    "<form>" +
+                    '<field name="body" widget="html" style="height: 100px"/>' +
+                    "</form>",
+                resId: 1,
+            });
+        },
+    },
+    function () {
+
+        QUnit.module("Banner");
+
+        QUnit.test("First element of o_editable is editable", async function (assert) {
```

Comment:
All test cases in this file is related to banner, if its confusing i will change it to something meaningful

Reply:
I mean it's not easy to tell from this title what it has to do with banner. I expect it to have a link with banner since it's in this file, but the title of this test is puzzling.

Reply:
So, is the test more like "Insert a banner"?

Reply:
Yes, it's to check whether the new banner is properly inserted or not.